modules/cas_cache/layer_cache_management.c

```diff
@@ -1992,39 +2012,45 @@ int cache_mngt_set_cache_mode(const char *cache_name, size_t name_len,
 {
 	ocf_cache_mode_t old_mode;
 	ocf_cache_t cache;
-	int result;
+	int result = 0;
 
 	result = ocf_mngt_cache_get_by_name(cas_ctx, cache_name,
 					name_len, &cache);
 	if (result)
 		return result;
 
-	result = _cache_mngt_lock_sync(cache);
-	if (result) {
-		ocf_mngt_cache_put(cache);
-		return result;
+	old_mode = ocf_cache_get_mode(cache);
+	if (old_mode == mode) {
+		printk(KERN_INFO "%s is in defined cache mode already", cache_name);
```

Comment:
I agree, however afaik there is no function to translate cache mode id to cache mode name. Since previous cache mode can always be retrieved from logs, I would leave this log as it is.